streamalert_cli/athena/helpers.py

```diff
@@ -57,10 +64,21 @@ def add_partition_statement(partitions, bucket, table_name):
             continue
 
         fmt_values.update(parts.groupdict())
+        partition_stmt = PARTITION_STMT.format(**fmt_values)
+
+        # If statement will be reset if it is about to reach the MAX_QUERY_LENGTH
+        if statement_len - len(partition_stmt) - len(statement) < 0:
+            results.append(' '.join(statement))
```

Comment:
you could make this a generator and have this `yield ' '.join(statement)` instead of storing them in a list that is returned

Reply:
Great idea!!